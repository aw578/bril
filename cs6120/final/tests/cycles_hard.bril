@main {
  a: int = const 0;
  one: int = const 1;
  two: int = const 2;
  ten: int = const 10;
  x: int = id a;
  y: int = id a;

.loop:
  p: int = add x one;
  q: int = add y two;
  c: bool = lt p ten;
  br c .then .end;

.then:
  x: int = id p;
  y: int = id q;
  jmp .loop;

.end:
  ret;
}