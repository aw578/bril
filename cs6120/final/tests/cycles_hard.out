Congruence classes for function main:
Class undef: ['p.init', 'c.init', 'q.init']
Class const 0: ['a.0']
Class const 1: ['one.0']
Class const 2: ['two.0']
Class const 10: ['ten.0']
Class id: ['y.0', 'x.0']
Class get: ['x.1', 'q.0', 'p.0', 'c.0', 'y.1']
Class add: ['p.1']
Class lt: ['c.1']
Class id_new: ['x.2']
Class add_new: ['q.1']
Class id_new_new: ['y.2']
